app/routes.js

```diff
@@ -157,10 +179,10 @@ export default (config) => {
         .sort()
         .map(mode => (mode === 'CITYBIKE' ? 'BICYCLE_RENT' : mode))
         .join(',')
-      : undefined,
+      : settings.modes,
       date: time ? moment(time * 1000).format('YYYY-MM-DD') : undefined,
```

Comment:
I tested this and it seems that custom mode set is ignored in routing. Disable train, save settings, go to front page and search a route from steisssi to pasila. Trains are included.